### PR TITLE
Fix exported TypeScript types

### PR DESCRIPTION
**Pull request by victorhmp**
#### What problem is this solving?

The exported TypeScript types were not taking into account the fact that the `Minicart` component can be used without any props being passed to it.

This is particularly relevant for Virtual Blocks.

#### How to test it?

Well, nothing should change visually, so this [Workspace](https://victormiranda--storecomponents.myvtex.com/) should look just fine.

### CODE REVIEW

**Review by klzns — APPROVED** [review]

File: react/Minicart.tsx [hunk 1]
```diff
@@ -15,14 +15,14 @@ import useCartIdPixel from './modules/useCartIdPixel'
...
+  variation?: MinicartVariationType
+  openOnHover?: boolean
+  linkVariationUrl?: string
+  maxDrawerWidth?: number | string
+  MinicartIcon?: React.ComponentType
+  drawerSlideDirection?: SlideDirectionType
+  quantityDisplay?: QuantityDisplayType
+  itemCountMode?: MinicartTotalItemsType
   backdropMode?: MaybeResponsiveValue<BackdropMode>
 }
```
Comment (Klynger) [inline react/Minicart.tsx]: Just a little suggestion, but it doesn't change anything

```typescript
interface MinicartComplateProps {
  variation: MinicartVariationType
  openOnHover: boolean
  linkVariationUrl: string
  maxDrawerWidth: number | string
  MinicartIcon: React.ComponentType
  drawerSlideDirection: SlideDirectionType
  quantityDisplay: QuantityDisplayType
  itemCountMode: MinicartTotalItemsType
  backdropMode: MaybeResponsiveValue<BackdropMode>
}

type MinicartProps = Partial<MinicardCompleteProps>
```

File: react/MinicartContext.tsx [hunk 2]
```diff
@@ -23,8 +23,8 @@ interface State {
 }
 
 interface Props {
-  openOnHover: boolean
-  variation: MinicartVariationType
+  openOnHover?: boolean
+  variation?: MinicartVariationType
```
Comment (Klynger) [inline react/MinicartContext.tsx:27]: Same as the above

File: react/modules/debouncedPixelHook.ts [hunk 3]
```diff
@@ -4,7 +4,9 @@ import debounce from 'debounce'
 
 export default function useDebouncedPush() {
   const { push } = usePixel()
-  const debouncedPush = useRef<(param: object) => void>(debounce(push, 600))
+  const debouncedPush = useRef<(param: Record<string, unknown>) => void>(
```
Comment (Klynger) [inline react/modules/debouncedPixelHook.ts]: You could name this type out of the component and use the name here, so the line wouldn't break

**Review by Klynger — COMMENTED** [review]
Just some nitpicks that you don't have to do, is just another way of doing things, but nothing that I suggest here is better or worse if you gonna do the changes is up to you

File: react/Minicart.tsx (hunk 1 above)
Comment (victorhmp) [reply react/Minicart.tsx]: Niice! It looks cleaner!
I don't even need to declare a new type, right? I could just say that the `Minicart` component has type `FC<Partial<MinicartProps>>`

**Pull request closed without merging**